experimental/Schemes/SpecOpen.jl

```diff
@@ -204,57 +204,50 @@ function intersect(
     check::Bool=true
   )
   X = ambient(U)
-  ambient_ring(X) === ambient_ring(Y) || error("Schemes can not be compared")
+  ambient_ring(U) === ambient_ring(Y) || error("schemes can not be compared")
   X == Y && return SpecOpen(Y, gens(U), check=check)
   if check && !issubset(Y, X)
     Y = intersect(Y, X)
   end
   return SpecOpen(Y, gens(U), check=check)
 end
 
-function intersect(
-    U::SpecOpen,
-    Y::Spec
-  )
-  return intersect(Y, U)
-end
+intersect(U::SpecOpen, Y::Spec) = intersect(Y, U)
 
 function intersect(
     U::SpecOpen,
     V::SpecOpen
   )
   X = ambient(U) 
-  is_canonically_isomorphic(X, ambient(V)) || error("ambient schemes do not coincide")
+  X == ambient(V) || error("ambient schemes do not coincide")
   return SpecOpen(X, [a*b for a in gens(U) for b in gens(V)])
 end
 
 function Base.union(U::T, V::T) where {T<:SpecOpen}
-  is_canonically_isomorphic(ambient(U), ambient(V)) || error("the two open sets do not lay in the same ambient scheme")
+  ambient(U) == ambient(V) || error("the two open sets are not contained in the same ambient scheme")
   return SpecOpen(ambient(U), vcat(gens(U), gens(V)))
 end
 
 function issubset(
-    Y::Spec, 
+    Y::Spec,
     U::SpecOpen
   )
+  ambient_ring(Y) === ambient_ring(U) || return false
+  issubset(Y, ambient(U)) || return false
   return one(OO(Y)) in ideal(OO(Y), gens(U))
 end
 
+
 function issubset(
     U::SpecOpen,
     Y::Spec
   ) 
-  for V in affine_patches(U)
-    issubset(V, Y) || return false
-  end
-  return true
+  return all(issubset(V, Y) for V in affine_patches(U))
 end
 
 function issubset(U::T, V::T) where {T<:SpecOpen}
```

Comment:
```suggestion
function issubset(U::SpecOpen, V::SpecOpen)
```

Reply:
Okay, I guess the equality of base rings are checked by equality of the ambient rings already.